Userland/Libraries/LibGfx/PortableFormatWriter.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2023, Lucas Chollet <lucas.chollet@free.fr>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/ByteBuffer.h>
+#include <LibGfx/Bitmap.h>
+
+namespace Gfx {
+
+// This is not a nested struct to work around https://llvm.org/PR36684
+struct PortableFormatOptions {
+    enum class Format {
+        Ascii,
+        Raw,
+    };
+
+    Format format = Format::Raw;
+    StringView comment = "Generated with SerenityOS - LibGfx."sv;
+};
+
+class PortableFormatWriter {
+public:
+    static ErrorOr<ByteBuffer> encode(Gfx::Bitmap const&, PortableFormatOptions options = PortableFormatOptions {});
```

Comment:
Please adopt the same `Options` naming as `BMPWriter`:

https://github.com/SerenityOS/serenity/blob/be4da57004ccc9a97e1e2ae1a28d101877b8f164/Userland/Libraries/LibGfx/BMPWriter.h#L27-L28

Reply:
Also, no `Gfx::` needed